test/adjoint.jl

```diff
@@ -589,4 +589,31 @@ using Test
   end
   reference_sol_cont = ForwardDiff.gradient(G_cont, p)
   @test easy_res_cont' ≈ reference_sol_cont rtol=1e-3
+
+  function rober(du,u,p,t)
+    y₁,y₂,y₃ = u
+    k₁,k₂,k₃ = p
+    du[1] = -k₁*y₁+k₃*y₂*y₃
+    du[2] =  k₁*y₁-k₂*y₂^2-k₃*y₂*y₃
+    du[3] =  y₁ + y₂ + y₃ - 1
+    nothing
+  end
+  M = [1. 0  0
+       0  1. 0
+       0  0  0]
+  f = ODEFunction(rober,mass_matrix=M)
+  p = [0.04,3e7,1e4]
+
+  Alg = Rosenbrock23 # Rodas5 fails because of interpolation weirdness. Maybe because this problem is very stiff? But that is very much unexpected.
```

Comment:
Rodas5 doesn't use the stiff-aware interpolation, it just uses 3rd order Hermite. So yeah, this could cause an issue.

Reply:
It does have a not implemented stiff-aware interpolation though, which might be good to add.